File: backend/MasApi/Models/Order.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MasApi.Models
{
    public class Order
    {
        [Key]
        public required Guid Id { get; set; }
        public required DateTime CreatedAt { get; set; }
        public required Guid CustomerId { get; set; }
        public Customer? Customer { get; set; }
        public required Guid SellerId { get; set; }
        public Seller? Seller { get; set; }
        public required Enums.Currency Currency { get; set; }

        public Invoice? Invoice { get; set; }
        public Delivery? Delivery { get; set; }
        public ICollection<Payment>? Payments { get; set; }
        public ICollection<OrderItem>? OrderProducts { get; set; }

        [NotMapped]
        public decimal TotalNetPrice => OrderProducts?.Sum(op => op.TotalNetPrice) ?? 0;
        [NotMapped]
        public decimal TotalVatAmount => OrderProducts?.Sum(op => op.TotalVatAmount) ?? 0;
        [NotMapped]
        public decimal TotalGrossPrice => OrderProducts?.Sum(op => op.TotalGrossPrice) ?? 0;
    }
}

File: frontend/src/types/domain/order.ts
import { Customer, Delivery, Invoice, OrderItem, Payment, Seller } from '.';
import { CustomerDto, DeliveryDto, InvoiceDto, OrderItemDto, PaymentDto, SellerDto } from '../dto';

export interface Order1 {
    id: string; // UUID
    createdAt: string; // ISO datetime
    customerId: string;
    sellerId: string;
    currency: string;
    status: string;
    totalNetPrice: number;
    totalGrossPrice: number;
    totalVatAmount: number;
}

export interface Order2 {
    id: string; // UUID
    createdAt: string; // ISO datetime
    customer: string;
    company: string;
    seller: string;
    status: string;
    deliveryId: string;
    invoiceNUmber: string;
}

export interface OrderTableData {
    lp: number;
    id: string;
    createdAt: string;
    customer: string;
    company: string;
    status: string;
    statusLabel: string;
    seller: string;
    invoiceNumber: string;
}

export interface OrderSummary {
    id: string; // UUID
    createdAt: string; // ISO datetime
    customer: Customer;
    status: string;
    currency: string;
    seller: Seller;
    totalNetPrice: number;
    totalVatAmount: number;
    totalGrossPrice: number;
    delivery: Delivery | null;
    invoice: Invoice | null;
    orderProducts: OrderItem[] | null;
    payments: Payment[] | null;
}

export interface OrderSummaryDto {
    id: string; // UUID
    createdAt: string; // ISO datetime
    customer: CustomerDto;
    status: string;
    currency: string;
    seller: SellerDto;
    totalNetPrice: number;
    totalVatAmount: number;
    totalGrossPrice: number;
    delivery: DeliveryDto | null;
    invoice: InvoiceDto | null;
    orderProducts: OrderItemDto[] | null;
    payments: PaymentDto[] | null;
}

export type OrderSortOption =
    | 'CREATED_DESC'
    | 'CREATED_ASC'
    | 'CUSTOMER_ASC'
    | 'CUSTOMER_DESC'
    | 'COMPANY_ASC'
    | 'COMPANY_DESC';
